app/components/provider_interface/offer_summary_list_component.rb

```diff
@@ -3,9 +3,14 @@ class OfferSummaryListComponent < ActionView::Component::Base
     include ViewHelper
     attr_reader :application_choice, :header
 
-    def initialize(application_choice:, header: 'Your offer')
+    def initialize(application_choice:, header: 'Your offer', extra_arguments: {})
       @application_choice = application_choice
       @header = header
+      @course_option_id = extra_arguments[:course_option_id]
+      @entry = extra_arguments[:entry]
+      @available_providers = extra_arguments[:available_providers]
```

Comment:
I might be misunderstanding the change here but wouldn't passing the change path remove a bit of path determining logic in this component? We don't seem to use these collections other than to work out which path to render.

Reply:
From 83673a382bd06c13c2bc22e6c5b6494d3ad91c83 onwards we also use the relevant counts to determine if the links should be shown. But if Adam agrees we may simplify this to just check `@entry`. Application choice + new course_option_id would then be sufficient to construct the change links. But it's an interesting point, whether the component should decide visibility and url of the links or just be given them.